app/controllers/breweries_controller.rb

```diff
@@ -142,34 +142,28 @@ def track_analytics
     ahoy.track action_name, params
   end
 
-  # Allow _ to be a separator
-  def format_query(query)
-    query.gsub('_', ' ')
-  end
-
   def validate_params
+    errors = []
     params.each do |key, value|
       next if %w[controller action].include?(key)
 
-      # NOTE: Throwing error --
-      # `undefined method `last' for #<ActionController::Parameters {} permitted: false>`
-      # https://sentry.io/share/issue/3ff46bbbcf0d4affa0de1f5cdd0fc461/
-      #
-      # if DISALLOWED_CHARACTERS.include?(value.last)
-      #   render body: "#{key} query parameter has improper value.", status: :bad_request
-      #   return
-      # end
+      # Convert underscores to spaces in params (for convenience)
+      value.gsub!('_', ' ') unless value.empty?
 
       case key
+      when 'query'
+        errors.push('Search cannot be blank.') if value.empty?
```

Comment:
Interesting this got past the middleware. I wonder what I was doing differently... Good job anyways!

Reply:
Wellllll, I never really tested it. It doesn't work 😆  Removing it 😄